### DISCUSSION TITLE
HmIP WRC6 - Keypress does not trigger Automation

### DISCUSSION THREAD

**Discussion by schnu240** [SukramJ/aiohomematic#1755] [Q&A] [answered]
Hello,
for quiet some time use the Keypress Service to read out the long_press from Homematic Switches to trigger automation.
I used to translate this approach to get the activation for the 6 button switch WRC6.
Unfortunately this does not work. What I have tried:
![image](https://github.com/user-attachments/assets/aad27875-03a5-4dfe-9c83-a03421bb63f4)

I created the necessary automation directly in my CCU3 and "learned" all the buttons in for the WRC6 as for my other Switches. 
I can read out the "long_press" via the Devtools in Home Assistant so this should work already.
![image](https://github.com/user-attachments/assets/8a17a935-44ab-48d5-92d4-811f1078df0b)
Do you have an idea what I am doing wrong?

**Comment by Baxxy13** [comment, marked as answer]
Man konnte doch in Automatisierungen direkt die verschiedenen "Key Press" - Events "zusammenklicken". 
Das geht nicht?

Ansonsten mal ohne Kanalnummer versuchen und dafür als subtype die Kanalnummer nehmen. 
Das sollte dann gehen, gerade getestet.

**Comment by schnu240** [comment]
Hi, danke für den Hinweis. Ich werde es morgen direkt mal probieren. Aktuell komme ich nicht an den Schalter weil alle schlafen :D

**Comment by schnu240** [comment]
Hey ja das war genau die Lösung. Vielen Vielen Dank :) Darauf wäre ich nie gekommen.
Kanalnummer als Subtype gewählt und dann hat es funktioniert.
